enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,641 @@
+     * isHostnameUnique: [success, failure, pending, error]
+     * isPlatformValid: [success, failure, pending, error]
+   * network:
+     * isConnected: [success, failure, pending, error]
+     * isMachineCidrDefined: [success, failure, pending, error]
+     * belongsToMachineCidr: [success, failure, pending, error]
+     * isApiVipConnected: [success, failure, pending, error]
+     * belongsToMajorityGroup: [success, failure, pending, error]
+     * isNtpSynced: [success, failure, pending, error]
+  * progress:
+    * currentStage: enum
+    * progressInfo: string
+    * stageStartedAt: string
+    * stageUpdatedAt: string
+ * connectivity: array of
+   * host
+   * l2Connectivity
+     * outgoingNic: string
+     * outgoingIpAddress: string
+     * remoteIpAddress: string
+     * remoteMac: string
+     * successful: boolean
+   * l3Connectivity
+     * outgoingNic: string
+     * remoteIpAddress: string
+     * successful: boolean
+ * apiVipConnectivity: boolean
+ * ntpSources: array of
+   * sourceName: string
+   * sourceState: [synced, combined, not_combined, error, variable, unreachable]
+
+
+#### REST API Access
+Some REST APIs need to be exposed in addition to the Kubernetes-native APIs
+described below.
+* Download ISO and PXE artifacts: These files must be available for download
+via HTTP, either directly by users or by a BMC.  Because BMCs do not pass
+authentication headers, the Assisted Service must generate some random URL or
+query parameter so that the ISO’s location isn’t easily guessable.
+* Agent APIs (near-term): Until a point where the agent creates and modifies
+AgentDrivenHostInstallation CRs itself, the agent will continue communicating
+with the service via REST APIs.  Currently the service embeds the user’s pull
+secret in the discovery ISO which the agent passes in an authentication
+header.  In this case the service can generate and embed some token which it
+can later validate.
+
+
+#### Hive Integration
+
+Hive has a [ClusterDeployment
+CRD](https://github.com/openshift/hive/blob/master/docs/using-hive.md#clusterdeployment)
+resource that represents a cluster to be created. It includes a "platform"
+section where platform-specific details can be captured. For an agent-based
+workflow, this section will include whatever information is necessary to get an
+ISO and boot hosts.
+
+A new field in the ClusterDeployment spec will optionally reference an
+AgentDrivenClusterInstallation resource, which describes everything that the
+agent-based install tooling needs to know about the cluster that it will be
+creating. Some or all of that new resource may be folded into ClusterDeployment
+in the future, but in the immediate term, keeping it separate will enable much
+quicker development.
+
+Hive or ACM will optionally orchestrate the booting of bare metal hosts by:
+
+1. Get an ISO URL from assisted operator.
+1. Use metal3's BareMetalHost to boot the live ISO on some hardware.
+
```

Comment:
From the perspective of hive, I still feel some  missing info for the workflow here... could we provide one example workflow here to understand the interactions here?

- a user creates the cluster deployment and agent driven cluster installation object, there’s a bunch of duplicate things in these objects like clustername etc..
- hive then uses these to figure out things like what version of open shift is required and then requests for an iso from the service
- Hive then uses the iso and some bare metal api to boot these hosts
- Hive then performs the matching of the hosts that report back to fulfill the cluster deployment requirements and then triggers the install??
- hive then keeps track of status of installation 

An high level workflow will help understand the next sections a little bit

Reply:
We reworked the document (and the design) a bit, hopefully it answers your questions.  Let us know if not.

Reply:
Some high-level workflow descriptions like this would make excellent introductory text for the proposal section around line 66, and give the overall context for the rest of these details.